### PR TITLE
Add Jenkinsfile and Dockerfile for AIO Jobs

### PR DESCRIPTION
**Pull request by hughsaunders**
This Jenkinsfile will be called by CIT Jenkins for PRs and Branch builds.
It will in turn call the RPC-AIO job to test swift/ceph/xenial/upgrades.
The results of the AIO builds will be added to the PR, with direct links.

Lint Jobs are executed in Docker, to isolate them from the CIT slave.
The Lint jobs are executed before AIOs to prevent wasting resources on
AIO builds when lint fails.

Connects rcbops/u-suk-dev#1313

### CODE REVIEW

**Comment by hughsaunders** [comment]
recheck_ceph

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM ubuntu
+RUN apt-get update
```
Comment (mkam) [inline Dockerfile]: Did we want to stay with apt-get update and install as separate commands?

**Comment by hughsaunders** [comment]
CIT ceph and xenial jobs failed on downloading ruby gems, retrying.

**Comment by hughsaunders** [comment]
skip ci

**Comment by hughsaunders** [comment]
recheck_ceph recheck_xenial

**Pull request merged**